python/fledge/common/logger.py

```diff
@@ -120,20 +117,51 @@ def setup(logger_name: str = None,
     logger.addHandler(handler)
     logger.propagate = propagate
 
-    @wraps(logger.error)
+    # Call error override
+    error_override(logger)
+    return logger
+
+
+def error_override(_logger: logging.Logger) -> None:
+    """Override error logger to print multi-line traceback and error string with newline
+    Args:
+        _logger: Logger Object
+
+    Returns:
+        None
+    """
+    # save the old logging.error function
+    __logging_error = _logger.error
+
+    @wraps(_logger.error)
     def error(msg, *args, **kwargs):
         if isinstance(msg, Exception):
+            """For example:
+                a) _logger.error(ex)
+                b) _logger.error(ex, "Failed to add data.")
+            """
             trace_msg = traceback.format_exception(msg.__class__, msg, msg.__traceback__)
             if args:
                 trace_msg[:0] = ["{}\n".format(args[0])]
             [__logging_error(line.strip('\n')) for line in trace_msg]
         else:
-            [__logging_error(m) for m in msg.splitlines()]
-
+            if isinstance(msg, str):
+                """For example:
+                    a) _logger.error(str(ex))
+                    b) _logger.error("Failed to log audit trail entry")
+                    c) _logger.error('Failed to log audit trail entry for code: %s', "CONCH")
+                    d) _logger.error('Failed to log audit trail entry for code: {log_code}'.format(log_code="CONAD"))
+                    e) _logger.error('Failed to log audit trail entry for code: {0}'.format("CONAD"))
+                    f) _logger.error("Failed to log audit trail entry for code '{}' \n{}".format("CONCH", "Next line"))
+                """
+                if args:
+                    msg = msg % args
```

Comment:
Add inline comment here in next PRs for log statements fixes

Reply:
Sure